chalk-engine/src/context.rs

```diff
@@ -94,7 +94,7 @@ pub trait ContextOps<I: Interner, C: Context<I>>: Sized + Clone + Debug {
     ) -> (
         C::InferenceTable,
         Substitution<I>,
-        Vec<InEnvironment<Constraint<I>>>,
+        Constraints<I>,
```

Comment:
I feel like it isn't worth using the interned `Constraints` here. We don't really intern anything else in `chalk-engine`. Maybe we should, but for consistency, let's just leave this as a `Vec`